{{ cookiecutter.project_name }}/{{ cookiecutter.package_name }}/scripts/train_test.py

```diff
@@ -1,11 +1,34 @@
 import pytest
+import pathlib
+
 from click.testing import CliRunner
 
-from {{cookiecutter.package_name}}.scripts.train import main
+from {{cookiecutter.package_name}}.scripts.train import main, _main
+
+_HERE = pathlib.Path(__file__).resolve().parent
 
 
 @pytest.mark.parametrize("config_file", [("configs/config.yml")])
-def test_train(config_file):
+def test_click_main(config_file):
+    """
+    This tests the _main() click runner AND ensures the main() function within it
+
+    If `result.exit_code` is non-zero, inserting the following lines right before the
+    `assert` statement can be very helpful for debugging. This will turn off `black`
+    formatting, then print the stack trace for the python function that failed.
+
+    ```
+    # fmt: off
```

Comment:
Why not just keep this always on?

Reply:
If you're worried about black, why not just write it out correctly formatted?

Reply:
Ah shoot. I set my editor to automatically `black`ify things on-save, so whenever I put any one-liners it would reformat them like
```
import traceback

traceback.whatever
```
which was insanely annoying